### ISSUE TITLE
moving up 2 or more lines at the same time crashes Zim

### ISSUE THREAD

**Issue by chronos000** [jaap-karssenberg/zim-desktop-wiki#261]
Zim allows to move a single line up or down with Ctrl+Up and Ctrl+Down. You can find these shortcuts in the "edit" menu.

These operations work well when there is no text selected, or when the selected text is within a selected line (no '\n' in the string).

If you select text spanning 2 or more lines, and try to use these shortcuts, Zim will crash immediately.

I found the curious case that if you select more than 2 lines, but you select them COMPLETELY (thus ending the selection exactly at a '\n'), then Zim wont crash, but it produces a strange behavior in which it removes some of the lines.

**## TL/DR;**

> **Description**
> Zim crashes when trying to move more than one line.
> 
> **how to reproduce:**
> Select 2 or more lines and press Ctrl+Up
> 
> **What should happen:**
> Zim moves all complete or partially selected lines upwards
> 
> **What happens:**
> Zim crashes

**Comment by Osndok** [comment]
Strange, I was all excite to discover a new shortcut, but I don't see this option... what is your platform & zim version?

Also, if you can you provide the diagnostic output from running zim with the "--debug" option, that might help expedite a fix.

**Comment by Osndok** [comment]
Oh, I found it... you have to enable the 'line sorter' plugin... I can reproduce the exception, but I also found sometimes the extra lines just... disappear.

```
ERROR: Exception during action: move_line_down
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/zim/actions.py", line 157, in do_activate
	self.get(instance, instance.class)()
  File "/usr/lib/python2.7/site-packages/zim/actions.py", line 140, in func
	self.func(instance, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/zim/plugins/linesorter.py", line 159, in move_line_down
	self.move_line(1)
  File "/usr/lib/python2.7/site-packages/zim/plugins/linesorter.py", line 133, in move_line
	iter = buffer.get_iter_at_line(target_line)
  File "/usr/lib/python2.7/site-packages/zim/gui/pageview.py", line 2731, in get_iter_at_line
	raise ValueError, 'Line number beyond the end of the buffer: %i' % line
ValueError: Line number beyond the end of the buffer: 9


ERROR: Exception during action: move_line_up
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/zim/actions.py", line 157, in do_activate
	self.get(instance, instance.class)()
  File "/usr/lib/python2.7/site-packages/zim/actions.py", line 140, in func
	self.func(instance, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/zim/plugins/linesorter.py", line 153, in move_line_up
	self.move_line(-1)
  File "/usr/lib/python2.7/site-packages/zim/plugins/linesorter.py", line 133, in move_line
	iter = buffer.get_iter_at_line(target_line)
  File "/usr/lib/python2.7/site-packages/zim/gui/pageview.py", line 2731, in get_iter_at_line
	raise ValueError, 'Line number beyond the end of the buffer: %i' % line
ValueError: Line number beyond the end of the buffer: 6
```

**Comment by elcste** [comment]
I hadn't seen the crash until playing with it some, but I often have it lose or mess up lines when I forget and try to move multiple lines :-/
